cli/flox-rust-sdk/src/providers/build.rs

```diff
@@ -689,4 +689,140 @@ mod tests {
+
+    fn log_file_created_and_persisted(mode: &str) {
+        let package_name = String::from("foo");
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [build.{package_name}]
+            sandbox = "{mode}"
+            command = """
+                echo "Hello, World!"
+                exit 42
+            """
+        "#};
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+
+        let output = assert_build_status(&flox, &mut env, &package_name, false);
+
+        let logfile_message_regex =
+            regex::Regex::new("ERROR: flox build failed \\(see (.+)\\)").unwrap();
+
+        let logfile_path = match logfile_message_regex.captures(&output.stderr) {
+            Some(captures) => Path::new(captures.get(1).unwrap().as_str()),
+            None => panic!("logfile path not found in stderr"),
+        };
+
+        assert!(logfile_path.exists(), "logfile not found: {logfile_path:?}");
+    }
+
+    #[test]
+    fn log_file_created_and_persisted_pure() {
+        log_file_created_and_persisted("pure");
+    }
+
+    #[test]
+    fn log_file_created_and_persisted_no_sandbox() {
+        log_file_created_and_persisted("off");
+    }
+
+    #[test]
+    fn dollar_out_persisted_no_sandbox() {
+        let package_name = String::from("foo");
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [build.{package_name}]
+            sandbox = "off"
+            command = """
+                echo "Hello, World!" >> $out
+                exit 42
+            """
+        "#};
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+
+        let output = assert_build_status(&flox, &mut env, &package_name, false);
+
+        let out_path_message_regex = regex::Regex::new("out=(.+?)\\s").unwrap();
+
+        let out_path = match out_path_message_regex.captures(&output.stdout) {
+            Some(captures) => Path::new(captures.get(1).unwrap().as_str()),
+            None => panic!("$out path not found in stdout"),
+        };
+
+        assert!(out_path.exists(), "out_path not found: {out_path:?}");
+
+        let out_content = fs::read_to_string(out_path).unwrap();
+        assert_eq!(out_content, "Hello, World!\n");
+    }
+
+    fn build_script_persisted(mode: &str, succeed: bool) {
+        let package_name = String::from("foo");
+
+        let command = if succeed {
+            r#"echo "Hello, World!" >> $out"#
+        } else {
+            "exit 42"
+        };
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [build.{package_name}]
+            sandbox = "{mode}"
+            command = '{command}'
+        "#};
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+
+        let output = assert_build_status(&flox, &mut env, &package_name, false);
```

Comment:
To fix these early failures for the success cases:
- https://github.com/flox/flox/actions/runs/11072117669/job/30765627816?pr=2138#step:7:660
- https://github.com/flox/flox/actions/runs/11072117669/job/30765627816?pr=2138#step:7:859
```suggestion
        let output = assert_build_status(&flox, &mut env, &package_name, succeed);
```